DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -200,6 +200,58 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Guild.Parse(response);
     }
 
+    public async IAsyncEnumerable<Data.Thread> GetGuildChannelThreadsAsync(
+        string channelId,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int currentOffset = 0;
+
+        var completeThreads = new List<JsonElement>();
+
+        while (true)
+        {
+            var url = new UrlBuilder()
+                .SetPath($"channels/{channelId}/threads/search")
+                .SetQueryParameter("offset", currentOffset.ToString())
+                .Build();
+            try
+            {
+                var response = await GetJsonResponseAsync(url, cancellationToken);
```

Comment:
You can use `TryGetJsonResponseAsync(...)` which returns `null` in case of an error status code. That way you don't have to `try`/`catch`.

Reply:
Resolved in f2c9bb1.